lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart

```diff
@@ -0,0 +1,228 @@
+/// instance is available, but simply not passed along to the component, those
+/// instance(s) will be used as the value for `props.store`/`props.actions`,
+/// even though the component itself may not make use of them internally.
+///
+/// In the case of a component that is rendered in a scope where a store/actions
+/// instance is not available, `null` will be used as the value for the prop(s).
+class RequiredFluxProps extends RecursiveAstVisitor with ClassSuggestor {
+  ResolvedUnitResult? _result;
+
+  static const fluxPropsMixinName = 'FluxUiPropsMixin';
+
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    var writesToFluxUiProps = false;
+    var actionsAssigned = false;
+    var storeAssigned = false;
+
+    DartType? fluxActionsType;
+    DartType? fluxStoreType;
+    final cascadeWriteEl = node.staticType?.element;
+    if (cascadeWriteEl is ClassElement) {
+      final maybeFluxUiPropsMixin = cascadeWriteEl.mixins
+          .singleWhereOrNull((e) => e.element.name == fluxPropsMixinName);
+      writesToFluxUiProps = maybeFluxUiPropsMixin != null;
+      fluxActionsType = maybeFluxUiPropsMixin?.typeArguments[0];
+      fluxStoreType = maybeFluxUiPropsMixin?.typeArguments[1];
+    }
+
+    final cascadingAssignments =
+        node.cascadeSections.whereType<AssignmentExpression>();
+    storeAssigned = cascadingAssignments.any((cascade) {
+      final lhs = cascade.leftHandSide;
+      return lhs is PropertyAccess && lhs.propertyName.name == 'store';
+    });
+    actionsAssigned = cascadingAssignments.any((cascade) {
+      final lhs = cascade.leftHandSide;
+      return lhs is PropertyAccess && lhs.propertyName.name == 'actions';
+    });
+
+    if (writesToFluxUiProps && !storeAssigned) {
+      storeAssigned = true;
+      final storeValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxStoreType) ?? 'null';
+      yieldNewCascadeSection(node, '..store = $storeValue');
+    }
+
+    if (writesToFluxUiProps && !actionsAssigned) {
+      actionsAssigned = true;
+      final actionsValue =
+          _getNameOfVarOrFieldInScopeWithType(node, fluxActionsType) ?? 'null';
+      yieldNewCascadeSection(node, '..actions = $actionsValue');
+    }
+  }
+
+  void yieldNewCascadeSection(CascadeExpression node, String newSection) {
+    final offset = context.sourceFile.getOffsetOfLineAfter(node.target.offset);
+    yieldPatch(newSection, offset, offset);
+  }
+
+  @override
+  Future<void> generatePatches() async {
+    _result = await context.getResolvedUnit();
+    if (_result == null) {
+      throw Exception(
+          'Could not get resolved result for "${context.relativePath}"');
+    }
+    _result!.unit.accept(this);
+  }
+}
+
+String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType? type) {
+  final inScopeVariableDetector = _InScopeVarDetector();
+  // Find top level vars
+  node.thisOrAncestorOfType<CompilationUnit>()?.accept(inScopeVariableDetector);
```

Comment:
It looks like this also finds out of scope variables, since it's starting at the compilation unit and recursing through everything (except function components).

For instance:
```dart
class MyComponent extends FluxUiComponent2... {
  render() {
    MyStore store; // Would pick this up
  }
}

void someOtherFunction() {
  Store store; // Would also pick this up
}

class MyModuleComponents {
  void someMethod() {
    MyStore store; // Would also pick this up
  }
}
```

This also only handles variable declarations, and not arguments.

I don't think Dart has any APIs for getting all the in-scope names, so we'll definitely just have to keep approximating this. But I think we should probably tweak this implementation to be more conservative.

How about something like this:
```diff
diff --git a/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart b/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
index 9d84eb9..694a83c 100644
--- a/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
+++ b/lib/src/dart3_suggestors/null_safety_prep/required_flux_props.dart
@@ -18,6 +18,7 @@ import 'package:analyzer/dart/ast/visitor.dart';
 import 'package:analyzer/dart/element/element.dart';
 import 'package:analyzer/dart/element/type.dart';
 import 'package:collection/collection.dart';
+import 'package:over_react_codemod/src/util.dart';
 import 'package:over_react_codemod/src/util/class_suggestor.dart';
 import 'package:over_react_codemod/src/util/offset_util.dart';
 
@@ -96,22 +97,46 @@ class RequiredFluxProps extends RecursiveAstVisitor with ClassSuggestor {
   }
 }
 
+class InScopeVariable {
+  final String name;
+  final DartType? type;
+
+  InScopeVariable(this.name, this.type);
+}
+
 String? _getNameOfVarOrFieldInScopeWithType(AstNode node, DartType? type) {
-  final inScopeVariableDetector = _InScopeVarDetector();
-  // Find top level vars
-  node.thisOrAncestorOfType<CompilationUnit>()?.accept(inScopeVariableDetector);
-  // Find vars declared in top-level fns (like `main()`)
-  node
-      .thisOrAncestorOfType<BlockFunctionBody>()
-      ?.visitChildren(inScopeVariableDetector);
+  final mostInScopeVariables = node.ancestors.expand((ancestor) sync* {
+    if (ancestor is FunctionDeclaration) {
+      // Function arguments
+      final element = ancestor.declaredElement;
+      if (element != null) {
+        yield* element.parameters.map((p) => InScopeVariable(p.name, p.type));
+      }
+    } else if (ancestor is Block) {
+      // Variables declared in the block (function body, if/else block, etc.)
+      yield* ancestor.statements
+          .whereType<VariableDeclarationStatement>()
+          .expand((d) => d.variables.variables)
+          .map((v) => InScopeVariable(v.name.lexeme, v.declaredElement?.type));
+    } else if (ancestor is ClassDeclaration) {
+      // Class fields
+      final element = ancestor.declaredElement;
+      if (element != null) {
+        yield* element.fields.map((f) => InScopeVariable(f.name, f.type));
+      }
+    } else if (ancestor is CompilationUnit) {
+      // Top-level variables
+      yield* ancestor.declarations
+          .whereType<TopLevelVariableDeclaration>()
+          .expand((d) => d.variables.variables)
+          .map((v) => InScopeVariable(v.name.lexeme, v.declaredElement?.type));
+    }
+  });
 
-  final inScopeVarName = inScopeVariableDetector.found
-      .firstWhereOrNull((v) {
-        final maybeMatchingType = v.declaredElement?.type;
-        return maybeMatchingType?.element?.name == type?.element?.name;
-      })
-      ?.declaredElement
-      ?.name;
+  final inScopeVarName = mostInScopeVariables.firstWhereOrNull((v) {
+    final maybeMatchingType = v.type;
+    return maybeMatchingType?.element?.name == type?.element?.name;
+  })?.name;
 
   final componentScopePropDetector = _ComponentScopeFluxPropsDetector();
   // Find actions/store in props of class components
@@ -147,26 +172,10 @@ bool _isFnComponentDeclaration(Expression? varInitializer) =>
     varInitializer is MethodInvocation &&
     varInitializer.methodName.name.startsWith('uiF');
 
-/// A visitor to detect in-scope store/actions variables (top-level and block function scopes)
-class _InScopeVarDetector extends RecursiveAstVisitor<void> {
-  final List<VariableDeclaration> found;
-
-  _InScopeVarDetector() : found = [];
-
-  @override
-  visitVariableDeclaration(VariableDeclaration node) {
-    // Don't visit function component declarations here since we visit them using the _ComponentScopeFluxPropsDetector
-    if (_isFnComponentDeclaration(node.initializer)) return;
-
-    if (node.declaredElement != null) {
-      found.add(node);
-    }
-  }
-}
-
```

Reply:
Addressed in https://github.com/Workiva/over_react_codemod/pull/268/commits/aea20f00a1cad1436a176e585a6341cc64c48800